File: update.py
#!/anaconda3/bin/python
import os
import cgi
import view
# import cgitb
# cgitb.enable()
print("Content-Type: text/html")
print()

files = os.listdir('data')

form = cgi.FieldStorage()
if 'id' in form:
    pageId = form["id"].value
    description = open('data/' + pageId, 'r').read()
else:
    pageId = 'Welcome'
    description = 'Hello, web'
print('''<!doctype html>
<html>
<head>
  <title>WEB2 - Welcome</title>
  <meta charset="utf-8">
</head>
<body>
  <h1><a href="index.py">WEB!!!!!</a></h1>
  <ol>
    {listStr}
  </ol>
  <a href="create.py">create</a>
      <form action="process_update.py" method="post">
          <input type="hidden" name="pageId" value="{form_default_title}">
          <p><input type="text" name="title" placeholder="Title" value="{form_default_title}"></p>
          <p><textarea rows="5" name="description" placeholder="Description">{form_default_description}</textarea></p>
          <p><input type="submit"></p>
      </form>
</body>
</html>
'''.format(
    title=pageId,
    desc=description,
    #    listStr=listStr,
    listStr=view.getList(),
    form_default_title=pageId,
    form_default_description=description))

# desc=description,
# listStr=view.getList(),
# update_link=update_link,
# delete_action=delete_action))

File: update.php
<?php
  require_once('lib/print.php');
  require_once('view/top.php');
 ?>

    <a href="create.php">create</a>
    <?php if(isset($_GET['id'])){?>
      <a href="update.php?id=<?=$_GET['id']?>">update</a>
      <?php } ?>

    <form action="update_process.php" method="post">
      <input type="hidden" name="old_title" value="<?=$_GET['id']?>">
      <p>
        <input type="text" name="title" size="30" value="<?php print_title();?>">
      </p>
      <p>
        <textarea name="description" rows="5" cols="70"><?php print_description();?></textarea>
      </p>
      <p>
        <input type="submit" value="SUBMIT">
      </p>
    </form>

<?php
  require('view/bottom.php');
 ?>
